### PR TITLE
Refactor IgxGrid sorting test suite (master)

### LINKED ISSUE #6997: Tests Refactoring: IgxGrid - Grid Sorting
Guidelines for test refactoring: https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular

Related to #6438

### PR DESCRIPTION
**Pull request by HristoP96**
Closes #6997

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
 
-describe('IgxGrid - Grid Sorting #grid', () => {
+
+function getCurrentCellFromGrid(grid, rowIndex, cellIndex) {
+    const gridRow = grid.rowList.toArray()[rowIndex];
+    const gridCell = gridRow.cells.toArray()[cellIndex];
+    return gridCell;
+}
+
+function getValueFromCellElement(cell) {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: You cab use GridFunctions.getValueFromCellElement(cell)

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 2]
```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
                 SortByParityComponent
             ],
             imports: [NoopAnimationsModule, IgxGridModule]
-        })
-        .compileComponents();
+        }).compileComponents();
     }));
 
     beforeEach(async(() => {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: can be changed to fakeAsync

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 3]
```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
...
-        expect(grid.getCellByColumn(0, currentColumn).value).toEqual(expectedResult);
-    });
-
-    it('Should sort grid descending by column name', () => {
-        const currentColumn = 'Name';
-        // Ignore case on sorting set to false
-        grid.sort({fieldName: currentColumn, dir: SortingDirection.Desc, ignoreCase: false});
-        fixture.detectChanges();
+            let expectedResult = 'ALex';
+            expect(grid.getCellByColumn(0, currentColumn).value).toEqual(expectedResult);
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: You can add directly expectation without constant

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 4]
```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
...
+            // Ignore case on sorting set to true
+            grid.sort({ fieldName: currentColumn, dir: SortingDirection.Asc, ignoreCase: true });
+            fixture.detectChanges();
 
-        expectedResult = 'Rick';
-        expect(grid.getCellByColumn(0, currentColumn).value).toEqual(expectedResult);
-        expectedResult = 'ALex';
-        expect(grid.getCellByColumn(grid.data.length - 1, currentColumn).value).not.toEqual(expectedResult);
+            expectedResult = 'ALex';
+            expect(grid.getCellByColumn(0, currentColumn).value).toEqual(expectedResult);
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: Same here just use:
expect(grid.getCellByColumn(0, currentColumn).value).toEqual('ALex');

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 5]
```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
...
 
-        const gridData = fixture.componentInstance.data;
-        const firstHeaderCell = GridFunctions.getColumnHeader('ID', fixture);
+        it('Should disable sorting feature when using NoopSortingStrategy.', () => {
+            grid.sortStrategy = NoopSortingStrategy.instance();
+            fixture.detectChanges();
 
-        GridFunctions.clickHeaderSortIcon(firstHeaderCell);
-        fixture.detectChanges();
+            const gridData = fixture.componentInstance.data;
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: gridData is not used.

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 6]
```diff
@@ -30,428 +42,378 @@ describe('IgxGrid - Grid Sorting #grid', () => {
...
+        });
 
-        const initialChips = fixture.debugElement.queryAll(By.directive(IgxChipComponent));
+        it('Should sort grid on sorting icon click when FilterRow is visible.', fakeAsync(/** Filtering showHideArrowButtons RAF */() => {
+            grid.allowFiltering = true;
+            fixture.detectChanges();
 
-        initialChips[0].nativeElement.click();
-        fixture.detectChanges();
+            const fistChip = ControlsFunction.getAllChipsFromAnElement(fixture.debugElement)[0];
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts]: You can change the:
  const fistChip = ControlsFunction.getAllChipsFromAnElement(fixture.debugElement)[0];

            UIInteractions.clickElement(fistChip);
            fixture.detectChanges();

            const filteringRow = fixture.debugElement.query(By.directive(IgxGridFilteringRowComponent));
            expect(filteringRow).toBeDefined();
to:
GridFunctions.clickFilterCellChipUI(fixture, 'Name');
expect(GridFunctions.getFilterRow(fixture)).toBeDefined();

File: projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts [hunk 7]
```diff
@@ -1846,6 +1846,10 @@ export class GridFunctions {
         }
     }
 
+    public static getHeaderCellByIndex(fix: ComponentFixture<any>, index: number) {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/test-utils/grid-functions.spec.ts]: You can use the method GridFunctions.getColumnHeader , where the name of the column is needed.

File: projects/igniteui-angular/src/lib/test-utils/controls-functions.spec.ts [hunk 8]
```diff
@@ -18,6 +19,10 @@ export class ControlsFunction {
        return chip.querySelector(CHIP_REMOVE_BUTTON);
     }
 
+    public static getAllChipsFromAnElement(element: DebugElement): DebugElement[] {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/test-utils/controls-functions.spec.ts:22]: Can you remove this as it is not used any more.

File: projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts [hunk 9]
```diff
@@ -1,27 +1,35 @@
...
 import { GridFunctions } from '../../test-utils/grid-functions.spec';
+import { GridDeclaredColumnsComponent, SortByParityComponent } from '../../test-utils/grid-samples.spec';
+import { ControlsFunction } from '../../test-utils/controls-functions.spec';
+import { UIInteractions } from '../../test-utils/ui-interactions.spec';
 
 const SORTING_ICON_ASC_CONTENT = 'arrow_upward';
 const SORTING_ICON_DESC_CONTENT = 'arrow_downward';
 
+
+function getCurrentCellFromGrid(grid, rowIndex, cellIndex) {
```
Comment (zdrawku) [inline projects/igniteui-angular/src/lib/grids/grid/grid.sorting.spec.ts:21]: Can you use `getCurrentCellFromGrid` that is imported from GridFunctions instead, we do not need to define another function with the same logic.

Example:
Change this:
`expect(GridFunctions.getValueFromCellElement(getCurrentCellFromGrid(grid, 0, 1))).toEqual('Jane');`
With:
`expect(GridFunctions.getValueFromCellElement(GridFunctions.getCurrentCellFromGrid(grid, 0, 1))).toEqual('Jane');`

**Pull request merged**